api/v1/ptpconfig_webhook.go

```diff
@@ -0,0 +1,153 @@
+)
+
+// log is for logging in this package.
+var ptpconfiglog = logf.Log.WithName("ptpconfig-resource")
+
+func (r *PtpConfig) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+type ptp4lConfSection struct {
+	options map[string]string
+}
+
+type ptp4lConf struct {
+	sections map[string]ptp4lConfSection
+}
+
+func (output *ptp4lConf) populatePtp4lConf(config *string, ptp4lopts *string) error {
+	lines := strings.Split(*config, "\n")
+	var currentSection string
+	output.sections = make(map[string]ptp4lConfSection)
+
+	for _, line := range lines {
+		if strings.HasPrefix(line, "[") {
+			currentSection = line
+			currentLine := strings.Split(line, "]")
+
+			if len(currentLine) < 2 {
+				return errors.New("Section missing closing ']'")
+			}
+
+			currentSection = fmt.Sprintf("%s]", currentLine[0])
+			section := ptp4lConfSection{options: map[string]string{}}
+			output.sections[currentSection] = section
+		} else if currentSection != "" {
+			split := strings.IndexByte(line, ' ')
+			if split > 0 {
+				section := output.sections[currentSection]
+				section.options[line[:split]] = line[split:]
+				output.sections[currentSection] = section
+			}
+		} else {
+			return errors.New("Config option not in section")
+		}
+	}
+	_, exist := output.sections["[global]"]
+	if !exist {
+		output.sections["[global]"] = ptp4lConfSection{options: map[string]string{}}
+	}
+
+	// When validating, add ptp4lopts to conf for fields we check
+	opts := strings.Split(*ptp4lopts, " ")
+	for index, opt := range opts {
+		if opt == "--summary_interval" && index < len(opts)-1 {
+			output.sections["[global]"].options["summary_interval"] = opts[index+1]
+		}
+	}
+	return nil
+}
+
+func (r *PtpConfig) validate() error {
+    if r.Name == "rejectme" {
+            return errors.New("Rejecting REJECTME")
+    }
+    profiles := r.Spec.Profile
+    for _, profile := range profiles {
+    	if *profile.Name == "rejectmetoo" {
+    		return errors.New("Rejecting invalid name too")
+    	}
+    	conf := &ptp4lConf{}
+    	conf.populatePtp4lConf(profile.Ptp4lConf, profile.Ptp4lOpts)
+
+    	// Validate that interface field only set in ordinary clock
+    	if *profile.Interface != "" {
+    		for section, _ := range conf.sections {
+    			if section != "[global]" {
+    				if section != ("[" + *profile.Interface + "]") {
+    					return errors.New("interface section " + section + " not allowed when specifying interface section")
+    				}
+    			}
+    		}
+    	}
+
+    	// Validate that summary_interval matches logSyncInterval
+    	summary_interval := ""
+    	logSyncInterval := ""
+    	for option, value := range conf.sections["[global]"].options {
+    		if option == "summary_interval" {
+    			summary_interval = value
+    		}
+    		if option == "logSyncInterval" {
+    			logSyncInterval = value
+    		}
+    	}
+    	if summary_interval != logSyncInterval {
```

Comment:
should we default it to -4 ?

Reply:
Where would -4 be passed to ptp4l as the default?  Would it be for both of them?

Reply:
ptp4lOpts: "-2  --summary_interval -4"

Reply:
Where is that actually set as the default?

Reply:
default in conf file is actually 0.